examples/cifar/main.py

```diff
@@ -14,45 +15,92 @@
 from bittensor import bittensor_pb2
 
 class CIFAR(bittensor.Synapse):
-        
     def __init__(self, config: bittensor.Config):
         super(CIFAR, self).__init__(config)
+        model_config = self.DPN26()
+        in_planes, out_planes = model_config['in_planes'], model_config['out_planes']
+        num_blocks, dense_depth = model_config['num_blocks'], model_config['dense_depth']
         
         # Main Network
-        self.conv1 = nn.Conv2d(3, 6, 5)
-        self.pool = nn.MaxPool2d(2, 2)
-        self.conv2 = nn.Conv2d(6, 16, 5)
-        self.fc1 = nn.Linear(16 * 5 * 5, 120)
-        self.fc2 = nn.Linear(120, 84)
-        self.fc3 = nn.Linear(84, 10)
-        
+        self.conv1 = nn.Conv2d(3, 64, kernel_size=3, stride=1, padding=1, bias=False)
+        self.bn1 = nn.BatchNorm2d(64)
+        self.last_planes = 64
+        self.layer1 = self._make_layer(in_planes[0], out_planes[0], num_blocks[0], dense_depth[0], stride=1)
+        self.layer2 = self._make_layer(in_planes[1], out_planes[1], num_blocks[1], dense_depth[1], stride=2)
+        self.layer3 = self._make_layer(in_planes[2], out_planes[2], num_blocks[2], dense_depth[2], stride=1)
+        self.layer4 = self._make_layer(in_planes[3], out_planes[3], num_blocks[3], dense_depth[3], stride=2)
+        self.linear = nn.Linear(out_planes[3] + (num_blocks[3] + 1) * dense_depth[3], 10)
+
         # Distill Network
-        self.dist_conv1 = nn.Conv2d(3, 6, 5)
-        self.dist_pool = nn.MaxPool2d(2, 2)
-        self.dist_conv2 = nn.Conv2d(6, 16, 5)
-        self.dist_fc1 = nn.Linear(16 * 5 * 5, 120)
-        self.dist_fc2 = nn.Linear(120, 84)
-        self.dist_fc3 = nn.Linear(84, 10)
+        self.dist_conv1 = nn.Conv2d(3, 64, kernel_size=3, stride=1, padding=1, bias=False)
+        self.dist_bn1 = nn.BatchNorm2d(64)
+        self.dist_layer1 = copy.deepcopy(self.layer1) 
+        self.dist_layer2 = copy.deepcopy(self.layer2) 
+        self.dist_layer3 = copy.deepcopy(self.layer3)
+        self.dist_layer4 = copy.deepcopy(self.layer4)
+
+    def set_batch_size(self, batch_size):
+        self.batch_size = batch_size
```

Comment:
Had to perform this mini-hack in checking the batch size because we flatten the input before we give it to the network.. this loses the shape information of the tensor, so while it runs well out of the network we still cannot reshape out input back on time to get the right batch_size back, so this is an extra module added  to help us set the batch size for the network to shape its output properly

Reply:
There's probably a better way of doing this, but for now this will do.